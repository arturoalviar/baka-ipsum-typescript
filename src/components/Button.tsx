import React, { MouseEvent } from 'react'
import styled from 'styled-components'
import theme from 'styled-theming'
import button from '../styles/components/button'

type Variant = 'primary' | 'default'

type ButtonProps = {
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void
  variant?: Variant
}

const { defaultStyles } = button
const primaryStyles = theme.variants('mode', 'variant', button.primaryStyles)

const Button: React.FC<ButtonProps> = ({
  variant = 'default',
  onClick,
  children,
}) => {
  const ButtonWrapper = styled.button<ButtonProps>`
    ${defaultStyles}
    ${primaryStyles}
  `
  return (
    <ButtonWrapper variant={variant} onClick={onClick}>
      {children}
    </ButtonWrapper>
  )
}

export default Button
