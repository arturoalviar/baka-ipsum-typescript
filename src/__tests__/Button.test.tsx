import React from 'react'
import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Button from '../components/Button'

describe('Button', () => {
  it('renders and matches button to snapshot', () => {
    const button = shallow(<Button />)
    expect(toJSON(button)).toMatchSnapshot()
  })

  it('runs a method when clicked', () => {
    const clickFunc = jest.fn()
    const button = mount(<Button onClick={clickFunc} />)
    button.find('button').simulate('click')

    expect(clickFunc).toHaveBeenCalled()

    button.unmount()
  })
})
