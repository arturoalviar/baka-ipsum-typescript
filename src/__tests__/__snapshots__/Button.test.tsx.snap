// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders and matches button to snapshot 1`] = `
<styled.button
  variant="default"
/>
`;
